inst/include/plant/models/ff16w_strategy.h

```diff
@@ -12,13 +12,39 @@ class FF16w_Strategy: public FF16_Strategy {
   typedef std::shared_ptr<FF16w_Strategy> ptr;
   FF16w_Strategy();
 
+  virtual std::vector<std::string> aux_names() {
+    std::vector<std::string> ret({
+      "competition_effect",
+      "net_mass_production_dt",
+      "evapotranspiration"
+    });
+    // add the associated computation to compute_rates and compute there
+    if (collect_all_auxiliary) {
+      ret.push_back("area_sapwood");
+    }
+    return ret;
+  }
+
+  double evapotranspiration_dt(double area_leaf) const;
+  double water_access(const FF16_Environment& environment,
+                      double height, double area_leaf_);
+
+  // temporarily treating all soil layers as the same
+  double compute_consumption(int resource_index) {
```

Comment:
Rather than passing an index, we can pass a pointer to the environment and loop over the number of resources (soil layers), returning a vector for each cohort.